hpat/_datetime_ext.h

```diff
@@ -3,644 +3,758 @@
-                                       npy_datetime *out) __UNUSED__;
-
-static void* np_datetime_date_array_from_packed_ints(uint64_t *dt_data,
-                                    int64_t n_elems, PyObject* dt_date_class) __UNUSED__;
-
-// void dt_to_timestamp(int64_t val, pd_timestamp *ts) {
-//     pandas_datetimestruct out;
-//     pandas_datetime_to_datetimestruct(val, PANDAS_FR_ns, &out);
-//     ts->year = out.year;
-//     ts->month = out.month;
-//     ts->day = out.day;
-//     ts->hour = out.hour;
-//     ts->minute = out.min;
-//     ts->second = out.sec;
-//     ts->microsecond = out.us;
-//     ts->nanosecond = out.ps * 1000;
-// }
-
-static inline PyObject* py_datetime_date_from_packed_int(uint64_t dt, PyObject* dt_date_class)
-{
-    uint64_t year = dt >> 32;
-    uint64_t month = (dt >> 16) & 0xFFFF;
-    uint64_t day = dt & 0xFFFF;
-    return PyObject_CallFunction(dt_date_class, "iii", year, month, day);
-}
-
-// given an array of packed integers for datetime.date (pd_timestamp_ext format),
-// create and return a pd.Series of datetime.date() objects
-static void* np_datetime_date_array_from_packed_ints(uint64_t *dt_data, int64_t n_elems, PyObject* dt_date_class)
-{
-#define CHECK(expr, msg) if(!(expr)){std::cerr << msg << std::endl; PyGILState_Release(gilstate); return NULL;}
-    auto gilstate = PyGILState_Ensure();
-
-    npy_intp dims[] = {n_elems};
-    PyObject* ret = PyArray_SimpleNew(1, dims, NPY_OBJECT);
-    CHECK(ret, "allocating numpy array failed");
-
-    for(int64_t i = 0; i < n_elems; ++i) {
-        PyObject * s = py_datetime_date_from_packed_int(dt_data[i], dt_date_class);
-        CHECK(s, "creating Python datetime.date object failed");
-        auto p = PyArray_GETPTR1((PyArrayObject*)ret, i);
-        CHECK(p, "getting offset in numpy array failed");
-        int err = PyArray_SETITEM((PyArrayObject*)ret, (char*)p, s);
-        CHECK(err==0, "setting item in numpy array failed");
-        Py_DECREF(s);
-    }
+        PANDAS_FR_D = 4,       // Days
+        PANDAS_FR_h = 5,       // hours
+        PANDAS_FR_m = 6,       // minutes
+        PANDAS_FR_s = 7,       // seconds
+        PANDAS_FR_ms = 8,      // milliseconds
+        PANDAS_FR_us = 9,      // microseconds
+        PANDAS_FR_ns = 10,     // nanoseconds
+        PANDAS_FR_ps = 11,     // picoseconds
+        PANDAS_FR_fs = 12,     // femtoseconds
+        PANDAS_FR_as = 13,     // attoseconds
+        PANDAS_FR_GENERIC = 14 // Generic, unbound units, can
+                               // convert to anything
+    } PANDAS_DATETIMEUNIT;
+
+    static int convert_datetimestruct_to_datetime(PANDAS_DATETIMEUNIT base,
+                                                  const pandas_datetimestruct* dts,
+                                                  npy_datetime* out) __UNUSED__;
+
+    static void*
+        np_datetime_date_array_from_packed_ints(uint64_t* dt_data, int64_t n_elems, PyObject* dt_date_class) __UNUSED__;
```

Comment:
May be split parameters like in function above?

Reply:
it fits into 120 symbols. That is why automatic code styler didn't change it

Reply:
I'm also like to see return type of the function on the same line as function name. It quite helpful if you `grep` sources. But auto codestyler has this intermediate style. If you know how to deprecate this please add this comment into PR #108